### PR TITLE
Support new context api

### PR DESCRIPTION
**Pull request by lfender6445**
This patch is to support new context API introduced to react

### CODE REVIEW

File: src/components/__tests__/Link-test.js [hunk 1]
```diff
@@ -28,42 +46,33 @@ describe('Link', () => {
...
-      const wrapper = shallow(<Link to={{ pathname: '/' }} onClick={onClick} />)
-      wrapper.simulate('click', event)
-      expect(onClick).to.have.been.calledWith(event)
+      const wrapper = setup({ to: { pathname: '/' }, onClick })
+      wrapper.find('Link').simulate('click', event)
+      const calledWithSyntheticEvent = onClick
+        .args[0][0]
+        .constructor
+        .name === 'SyntheticEvent'
+      expect(calledWithSyntheticEvent).to.equal(true)
```
Comment (lfender6445) [inline src/components/__tests__/Link-test.js]: i am really not sure why this stopped working as it was written before. the event handler has not changed at all

**Comment by lfender6445** [comment]
Hmm. I have not been able to try this out locally unfortunately - hitting what appears to be a collision on the react versions

```{ renderingError:
   { Invariant Violation: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
   1. You might have mismatching versions of React and the renderer (such as React DOM)
   2. You might be breaking the Rules of Hooks
   3. You might have more than one copy of React in the same app
   See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
       at invariant (/Users/lfender/source/react-redux-router/node_modules/react/cjs/react.development.js:88:15)
       at resolveDispatcher (/Users/lfender/source/react-redux-router/node_modules/react/cjs/react.development.js:1436:28)
       at useMemo (/Users/lfender/source/react-redux-router/node_modules/react/cjs/react.development.js:1491:20)
       at ConnectFunction (/Users/lfender/source/react-redux-ro```

File: package.json [hunk 2]
```diff
@@ -17,17 +17,17 @@
...
   "devDependencies": {
-    "@babel/cli": "^7.4.3",
-    "@babel/core": "^7.4.3",
-    "@babel/plugin-proposal-class-properties": "^7.4.0",
-    "@babel/plugin-proposal-object-rest-spread": "^7.4.3",
-    "@babel/plugin-transform-destructuring": "^7.4.3",
-    "@babel/plugin-transform-modules-commonjs": "^7.4.3",
-    "@babel/plugin-transform-runtime": "^7.4.3",
-    "@babel/preset-env": "^7.4.3",
+    "@babel/cli": "^7.4.4",
```
Comment (lfender6445) [inline package.json]: ive updated all the dev deps here as well

File: src/components/Link.js [hunk 3]
```diff
@@ -1,25 +1,23 @@
 import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
+import { ReactReduxContext } from 'react-redux'
```
Comment (tadjohnston) [inline src/components/Link.js]: 👍 nice!

File: package.json [hunk 4]
```diff
@@ -36,30 +36,29 @@
...
   },
   "peerDependencies": {
-    "react": "^16.5.0",
-    "react-redux": "^4.0.0 || ^5.0.0",
-    "redux": "^3.7.2"
+    "react": "16.8.6"
   },
   "engines": {
-    "node": ">=8.0.0"
+    "node": ">=11.9.0"
```
Comment (mneumark) [inline package.json]: Should we just be updating this all the way to 12.2.0 since both ag and rent are now on that?

Comment (lfender6445) [reply package.json]: ag rolled back to 11 when we were diagnosing memory issues

Comment (mneumark) [reply package.json]: ah, thats fine then.

**Pull request merged**